### DISCUSSION TITLE
Feature: create umami report from api

### DISCUSSION THREAD

**Discussion by boly38** [umami-software/umami#1256] [Ideas]
Hi
 I created a proof of concept : using github action to prepare an umami daily report based on api call

 https://github.com/boly38/action-umami-report

this is a very first mvp

I plan to extract api client part into dedicated repo if no one exist (if such client already exist, please let me know)

feedbacks are welcome.
regards

**Comment by rohandebsarkar** [comment]
Another alternative for email reports could be to create a new API endpoint, such as `/api/report` by creating a file `pages/api/report.js`.

GitHub actions can be used to invoke the API at regular intervals. When the API is invoked, it sends an email report to a particular email address.

Although, I also like the approach that you are using. Your approach is more modular and can be used to generate different type of reports.

**Comment by boly38** [comment]
[umami-api-client](https://github.com/boly38/umami-api-client) v0.0.0 is [out](https://www.npmjs.com/package/umami-api-client) 😄

**Comment by boly38** [comment]
action-umami-report v1.1 is out

**Comment by boly38** [comment]
v1.4 has been [published into marketplace](https://github.com/marketplace/actions/create-umami-report)

Note that this actions may be an answer for ticket umami-software/umami#920 and umami-software/umami#924

**Comment by imjuniper** [comment]
Hey @boly38! I made an API client too, quite a bit inspired by yours honestly. I made it in TypeScript however, and I removed the need to pass the auth data on every request. I also added a unauthenticated client, that only supports the collect endpoint. It only has a single dependency, `axios`. I wanted to go with no dependency, but global fetch isn't super widespread yet. `axios` is quite a bit smaller than `node-fetch`, so the overall package is much smaller.
